lib/validation/validators/requiredIf.ts

```diff
@@ -0,0 +1,19 @@
+import { required } from './required'
+
+export type RequiredIfCondition = boolean | string | (() => boolean) | (() => Promise<boolean>)
+
+export async function requiredIf(value: unknown, condition: RequiredIfCondition): Promise<boolean> {
+  if (typeof condition === 'boolean' && condition) {
+    return required(value)
+  }
+
+  if (typeof condition === 'string' && condition) {
+    return required(value)
+  }
```

Comment:
I'm not sure what kind of string is supposed to make the field required.

Reply:
The condition check should be the same as `helpers.req` I think 👀  So for example empty string should not match the condition.

Reply:
so, something like this?

```ts
return (isString(value) || isBoolean(value) || (await condition())
  ? required(value)
  : true
}
```

Reply:
`isString` seems to return `true` for empty string.

```ts
export function isString(value: unknown): value is string {
  return typeof value === 'string'
}
```

So, current check logic is fine, or should I write like `isString(condition) && condition` (adding `isBoolean` and `isFunction` utils too for other branches)?

Reply:
Nah you can leave it as is 👍